pyOCD/target/target_kinetis.py

```diff
@@ -93,26 +93,27 @@ def init(self):
         if isLocked:
             return
 
-        # Prevent the target from resetting if it has invalid code
-        self.transport.writeAP(MDM_CTRL, MDM_CTRL_DEBUG_REQUEST | MDM_CTRL_CORE_HOLD_RESET)
-        while self.transport.readAP(MDM_CTRL) & (MDM_CTRL_DEBUG_REQUEST | MDM_CTRL_CORE_HOLD_RESET) != (MDM_CTRL_DEBUG_REQUEST | MDM_CTRL_CORE_HOLD_RESET):
+        if self.halt_on_connect:
```

Comment:
I like the ability to attach to a running target, but I'm not sure of the stability implications it will have.  If the target is running there are a number of things it can do that could cause a TransferError - like performing a software reset or entering a sleep mode.  I think this should work 99% of the time if there is valid code loaded though.